pygeoapi/filters.py

```diff
@@ -0,0 +1,290 @@
+from operator import and_, or_, add, sub, mul, truediv
+# from datetime import datetime, timedelta
+# from functools import reduce
+
+
+def combine(sub_filters, combinator="AND"):
+    """ Combine filters using a logical combinator
+
+        :param sub_filters: the filters to combine
+        :param combinator: a string: "AND" / "OR"
+        :type sub_filters: list of filtered dict 
+        :return: the combined filter
+        :rtype: filtered dict
+    """
+    # check for sub_filters being the evaluation result of multiple sub_filters
+    assert isinstance(sub_filters, tuple)
+
+    # check for sub_filter being the list of filtered dataset
+    for sub_filter in sub_filters:
+        assert isinstance(sub_filter, list)
+
+    # check for appropriate type of combinator
+    assert combinator in ("AND", "OR")
+
+    mapping_list = []
+    intersection = []
+    union = []
+    
+    for row in sub_filters[0]:
+        if row in sub_filters[1]:
+            intersection.append(row)
+        else:
+            union.append(row)
+    
+    # perform combination operation
+    if combinator == "AND":
+        mapping_list=intersection
+    else:
+        mapping_list = union + sub_filters[1]
+        
+    return mapping_list
+
+
+def negate(sub_filter):  # TODO!!
+    """ Negate a filter, opposing its meaning.
+
+        :param sub_filter: the filter to negate
+        :type sub_filter: query
+        :return: the negated filter
+        :rtype: 
+    """
+
+# Comparison operators dictionary
+OP_TO_COMP = {
+    "<": "<",
+    "<=": "<=",
+    ">": ">",
+    ">=": ">=",
+    "<>": "!=",
+    "=": "=="
+}
+
+
+def compare(lhs, rhs, op, mapping_choices=None, field_mapping=None):
```

Comment:
can you use explicit variable names?

Reply:
like what?

Reply:
what does `lhs` stand for?

Reply:
`lhs` is for Left hand side of the operator and `rhs` is for right hand side of the operator